pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1131,6 +1038,24 @@ async def parse_packages(self, primary_xml, filelists_xml, other_xml, file_exten
                 await packages_pb.aincrement()
                 await self.put(dc)
 
+            if settings.RPM_ITERATIVE_PARSING:
+                if settings.RPM_USE_NEW_ITERATIVE_PARSER:
+
+                    @async_to_sync
+                    async def _package_callback(pkg):
+                        if not skip_srpms or pkg.arch != "src":
+                            await on_package(pkg)
```

Comment:
This part feels hairy and I don't like it, but it's necessary because the async code is now being driven by this synchronous code rather than the other way around.  It's an unfortunate limitation of this internal iterator API but I see why they had to design the API that way.